Makefile.shared

```diff
@@ -103,7 +103,7 @@ CALC_VERSIONS=	\
 LINK_APP=	\
   ( $(SET_X);   \
     LIBDEPS="$${LIBDEPS:-$(LIBDEPS)}"; \
-    LDCMD="$${LDCMD:-$(CC)}"; LDFLAGS="$${LDFLAGS:-$(CFLAGS) $(LDFLAGS)}"; \
+    LDCMD="$${LDCMD:-$(CC)}"; LDFLAGS="$${LDFLAGS:-$(LDFLAGS)}"; \
```

Comment:
$(CFLAGS) cannot remain there, there are cc flags the linker simply doesn't accept.

Compiling for SAW happens to expose this.

If there are flags that are common to ld and cc, then we should have a variable for those common flags.